src/routes/Manual.svelte

```diff
@@ -211,6 +241,62 @@ $: drawableHexagons = hexagons.map(({id, row, col}) => {
     return {id, x, y, width, height, points: points.join(' '), color};
 });
 
+function buildCommandBlocks(active){
+    let actuator = active;
+    console.log(actuator);
+    let shiftActuator; 
+    let binActuator;
+    if (actuator <= 31) {
+        shiftActuator = Math.abs(1 << actuator); 
+        binActuator = shiftActuator.toString(2).padStart(32,'0'); 
+        block0_31[0] = binActuator.slice(24,32);
+        block0_31[1] = binActuator.slice(16,24);
+        block0_31[2] = binActuator.slice(8,16);
+        block0_31[3] = binActuator.slice(0,8); //string representation of binary command 
+
+        block32_63 = [0,0,0,0];
+        block64_95 = [0,0,0,0];
+        block96_127 = [0,0,0,0];
+        // block0_31[3] = parseInt(binActuator.slice(0,8),2); //decimal representation of binary command
+    } else if (actuator >= 32 && actuator < 63){
+        shiftActuator = Math.abs(1 << (actuator - 32)); 
+        binActuator = shiftActuator.toString(2).padStart(32,'0');
+        block32_63[0] = binActuator.slice(24,32);
+        block32_63[1] = binActuator.slice(16,24);
+        block32_63[2] = binActuator.slice(8,16);
+        block32_63[3] = binActuator.slice(0,8);
+
+        block0_31 = [0,0,0,0];
+        block64_95 = [0,0,0,0];
+        block96_127 = [0,0,0,0];
+    } else if (actuator >= 64 && actuator < 95){
+        shiftActuator = Math.abs(1 << (actuator - 64)); 
+        binActuator = shiftActuator.toString(2).padStart(32,'0');
```

Comment:
It would be nice if there was a short helper function that is tested. We may want to use such a utility for grabbing bytes elsewhere. It seems like with the shifting that you did, we may be able to have it assign all the blocks. without the if statements.

```
function getBytesForActuator(actuator) {
   const shiftActuator = ...;
   const binActuator = ...;
   return [...Array(8).keys()].map(i => binActuator.slice(i * 8, (i+1) * 8)).reversed();
}

const [b0, b1, b2, b3, ...rest] = getNumericActuator(actuator);
block0_31 = [b0, b1, b2, b3]

```

Reply:
I added the helper function, and agree that it may be helpful in the future. At the moment it does still use the if statements, but your method of slicing into an array did help clean it up at least. I watched the actuator command test while working on the helper function so I confirmed the helper function works as before.